src/IssueViz.UnitTests/Editor/QuickActions/QuickFixes/QuickFixSuggestedActionTests.cs

```diff
@@ -30,9 +37,128 @@ public class QuickFixSuggestedActionTests
+
+            var span2 = new Span(2, 20);
+            var snapshotSpan2 = new SnapshotSpan(snapShot.Object, span2);
+            var editVisualization2 = new Mock<IQuickFixEditVisualization>();
+            editVisualization2.Setup(e => e.Edit.Text).Returns("edit2");
+            editVisualization2.Setup(e => e.Span).Returns(snapshotSpan2);
+
+            var span3 = new Span(3, 30);
+            var snapshotSpan3 = new SnapshotSpan(snapShot.Object, span3);
+            var editVisualization3 = new Mock<IQuickFixEditVisualization>();
+            editVisualization3.Setup(e => e.Edit.Text).Returns("edit3");
+            editVisualization3.Setup(e => e.Span).Returns(snapshotSpan3);
+
+            var quickFixViz = new Mock<IQuickFixVisualization>();
+            quickFixViz.Setup(x => x.EditVisualizations).Returns(new List<IQuickFixEditVisualization> { editVisualization1.Object, editVisualization2.Object, editVisualization3.Object, });
+
+
+            var textEdit = new Mock<ITextEdit>(MockBehavior.Strict);
+            var textBuffer = new Mock<ITextBuffer>(MockBehavior.Strict);
+
+            var sequence = new MockSequence();
+
+            textBuffer.InSequence(sequence).Setup(t => t.CreateEdit()).Returns(textEdit.Object);
+            textEdit.InSequence(sequence).Setup(t => t.Replace(span1, "edit1")).Returns(true);
+            textEdit.InSequence(sequence).Setup(t => t.Replace(span2, "edit2")).Returns(true);
+            textEdit.InSequence(sequence).Setup(t => t.Replace(span3, "edit3")).Returns(true);
+            textEdit.InSequence(sequence).Setup(t => t.Apply()).Returns(Mock.Of<ITextSnapshot>());
+
+            var testSubject = new QuickFixSuggestedAction(quickFixViz.Object, textBuffer.Object);
+            testSubject.Invoke(CancellationToken.None);
+
+            textBuffer.Verify(tb => tb.CreateEdit(), Times.Once(), "CreateEdit should be called once");
+            textEdit.Verify(tb => tb.Replace(It.IsAny<Span>(), It.IsAny<string>()), Times.Exactly(3), "Replace should be called three time");
+            textEdit.Verify(tb => tb.Apply(), Times.Once(), "Apply should be called once");
+
+        }
+
+        [TestMethod]
+        public void CancellationToken_Cancels()
+        {
+            var quickFixViz = new Mock<IQuickFixVisualization>();
+            var textBuffer = new Mock<ITextBuffer>();
+            var textEdit = new Mock<ITextEdit>();
+            textBuffer.Setup(tb => tb.CreateEdit()).Returns(textEdit.Object);
+
+            var testSubject = new QuickFixSuggestedAction(quickFixViz.Object, textBuffer.Object);
+            CancellationTokenSource tokenSource = new CancellationTokenSource();
+            quickFixViz.SetupGet(x => x.EditVisualizations).Returns(Array.Empty<IQuickFixEditVisualization>());
+
+            var task = Task.Factory.StartNew(() =>
+            {
+                _ = Task.Delay(100);
+                testSubject.Invoke(tokenSource.Token);
+            });
+
+            tokenSource.Cancel();
```

Comment:
I think you can simplify it and just pass a token that is already cancelled

Reply:
I think this would be a more realistic scenario.

Reply:
We are only testing the token at the start, if it was cancelled after the IF then we wouldn't care anyway.

Reply:
@ugras-ergun-sonarsource I think without the threading the test can be simplified even further, for example: https://github.com/SonarSource/sonarlint-visualstudio/blob/c2de59e802f1a270168b139d3432ba8ceffd5ec7/src/IssueViz.UnitTests/Editor/QuickActions/QuickFixes/QuickFixSuggestedActionTests.cs#L127